flocker/node/test/test_script.py

```diff
@@ -558,6 +618,34 @@ def deployer_factory(node_uuid, hostname, cluster_uuid):
         agent.get_service(reactor, options)
         self.assertIn(spied[0], get_all_ips())
 
+    def test_deployer_factory_called_with_hostname(self):
+        """
+        If the configuration does specify a ``hostname``,
+        ``AgentServiceFactory.main`` calls its ``deployer_factory`` with one of
+        the hostname.
+        """
+        spied = []
+
+        def deployer_factory(node_uuid, hostname, cluster_uuid):
+            spied.append(hostname)
+            return object()
+
+        reactor = MemoryCoreReactor()
+        options = DatasetAgentOptions()
+
+        config = yaml.safe_load(self.config.getContent())
+        config["hostname"] = "hostname.local"
+        self.config.setContent(yaml.safe_dump(config))
```

Comment:
This is a bit ugly, but I think the entire `flocker.node.script` needs to be refactored so there is less duplication. And this then probably won't need to be reading a file.